R/uniDimCommon.R

```diff
@@ -13,17 +13,36 @@
 }
 
 
-.checkErrors <- function(dataset, options) {
-
-
-  .hasErrors(dataset = dataset,
-             type = c("infinity", "variance", "observations"),
-             observations.amount = " < 3",
-             infinity.target = options$variables,
-             variance.target = options$variables,
-             observations.target = options$variables,
-             exitAnalysisIfErrors = TRUE)
+.checkErrors <- function(dataset, options, Bayes = FALSE) {
+
+  if (Bayes) {
+    # check for sensible MCMC values
+    .checkMCMCBounds <- function() {
+      # are the remaining samples after burnin and thinning larger than 2?
+      if (ceiling((options[["noSamples"]] - options[["noBurnin"]]) / options[["noThin"]]) < 2) {
+        return(gettext("Too few MCMC samples will remain after the removal of the burnin samples and thinning,
+                     please increase No.samples."))
+      }
+      return(NULL)
+    }
 
+    .hasErrors(dataset = dataset,
+               type = c("infinity", "variance", "observations"),
+               observations.amount = " < 3",
+               infinity.target = options$variables,
+               variance.target = options$variables,
+               observations.target = options$variables,
+               custom = .checkMCMCBounds,
+               exitAnalysisIfErrors = TRUE)
+  } else {
+    .hasErrors(dataset = dataset,
+               type = c("infinity", "variance", "observations"),
+               observations.amount = " < 3",
+               infinity.target = options$variables,
+               variance.target = options$variables,
+               observations.target = options$variables,
+               exitAnalysisIfErrors = TRUE)
+  }
```

Comment:
you can unify these statements into something like this:

```r
.checkErrors <- function(dataset, options, Bayes = FALSE) {

  if (Bayes) {
    # check for sensible MCMC values
    custom <- list(checkMCMCBounds = function() {
      # are the remaining samples after burnin and thinning larger than 2?
      if (ceiling((options[["noSamples"]] - options[["noBurnin"]]) / options[["noThin"]]) < 2) {
        return(gettext("Too few MCMC samples will remain after the removal of the burnin samples and thinning,
                     please increase No.samples."))
      }
      return(NULL)
    })
  } else {
    custom <- NULL
  }

  .hasErrors(dataset = dataset,
             type = c("infinity", "variance", "observations"),
             observations.amount = " < 3",
             infinity.target = options$variables,
             variance.target = options$variables,
             observations.target = options$variables,
             custom = custom,
             exitAnalysisIfErrors = TRUE)
}
```
`.hasErrors` has an explicit argument for `custom` with default `NULL`.

Reply:
Do I need 


> ```r
>     custom <- list(checkMCMCBounds = function() {
> ```r
 the list argument here? It also works without it on my computer.

Reply:
you don't need that
```r
custom <- function() {
```
would also work. But as soon as you want to do multiple custom checks, they need to be put in a list.